lib/at_host/at_host.c

```diff
@@ -53,31 +53,37 @@ enum select_uart {
 
 static enum term_modes term_mode;
 static struct device *uart_dev;
-static int at_socket_fd = INVALID_DESCRIPTOR;
-static struct pollfd fds[1];
-static int nfds;
 static u8_t at_buf[AT_MAX_CMD_LEN];
 static size_t at_buf_len;
-static struct k_work at_cmd_send_work;
-static struct k_thread socket_thread;
-static K_THREAD_STACK_DEFINE(socket_thread_stack, THREAD_STACK_SIZE);
-static struct k_mutex socket_mutex;
-
-
+static struct k_work cmd_send_work;
 static const char termination[3] = { '\0', '\r', '\n' };
 
-static void at_cmd_send(struct k_work *work)
+static void cmd_send(struct k_work *work)
 {
-	int bytes_sent;
+	int err;
 
 	ARG_UNUSED(work);
 
-	k_mutex_lock(&socket_mutex, K_FOREVER);
-	bytes_sent = send(at_socket_fd, at_buf, at_buf_len, 0);
-	k_mutex_unlock(&socket_mutex);
+	err = at_cmd_write(at_buf);
+	if (err != AT_CMD_OK) {
+		LOG_ERR("Could not send AT command to modem: %d", err);
+	}
 
-	if (bytes_sent <= 0) {
-		LOG_ERR("Could not send AT command to modem: %d", bytes_sent);
+	switch (err) {
+	case AT_CMD_OK:
+		printk("OK\n");
```

Comment:
These are not logs, they are the command return codes from the modem. Because the at_interface driver parses these command return codes I must add them back into the consol to keep the at_host library behaviour as it was before.

Reply:
In that case, `uart_poll_out(uart_dev, ...)` should be used to send that data. There is no guarantee that `printk` will interface the UART that is configured by this module, actually it's very likely it won't, and push the data over RTT for instance.

Reply:
Ok